src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -84,25 +105,73 @@ std::optional<std::array<double, Dim>> Translation<Dim>::inverse(
              << function_of_time.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
-  double radial_function_value = 1.0;
-  if (f_of_r_ != nullptr) {
-    const double root = root_finder(target_coords - center_, function_of_time);
-    radial_function_value = (*f_of_r_)(root);
-  }
-  for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+  if (std::isnan(inner_radius_)) {
+    double radial_function_value = 1.0;
+    if (f_of_r_ != nullptr) {
+      const double root =
+          root_finder(target_coords - center_, function_of_time);
+      radial_function_value = (*f_of_r_)(root);
+    }
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+    }
+    return result;
+  } else {
+    auto target_radius = magnitude(target_coords);
+    double non_translated_radius = 0.;
+    for (size_t i = 0; i < Dim; i++) {
+      non_translated_radius +=
+          square(gsl::at(target_coords, i) - function_of_time[i]);
+    }
+    non_translated_radius = sqrt(non_translated_radius);
+    if (target_radius >= outer_radius_) {
+      return result;
+    } else if (non_translated_radius <= inner_radius_) {
+      for (size_t i = 0; i < Dim; i++) {
+        gsl::at(result, i) -= function_of_time[i];
+      }
+    } else {
+      // Solving a quadratic of known quantities.
+      double a = 0.;
+      double b = 0.;
+      double c = 0.;
+      for (size_t i = 0; i < Dim; i++) {
+        a += square(function_of_time[i]);
+        b -= 2 * function_of_time[i] * gsl::at(target_coords, i);
+        c += square(gsl::at(target_coords, i));
+      }
+      a -= square(outer_radius_ - inner_radius_);
+      b += 2 * outer_radius_ * (outer_radius_ - inner_radius_);
+      c -= square(outer_radius_);
+      auto roots = real_roots(a, b, c);
+      double radial_falloff_factor = 0.;
+      if (roots.value()[0] >= 0.0 && roots.value()[0] <= 1.0) {
+        ASSERT(roots.value()[1] < 0.0 || roots.value()[1] > 1.0,
```

Comment:
We use `and` and `or` instead of `&&` and `||`. It's just easier for people to read. Same for the rest of this PR.